### PR TITLE
Remove finalizers handling from api handler

### LINKED ISSUE #10376: Remove finalizer handling for ExternalClusters from API
### What happened?
Currently the KKP API is the only place that sets finalizers like `ExternalClusterKubeOneNamespaceCleanupFinalizer`. This must not happen, however, as ExternalCluster objects can also be created by `kubectl apply`ing them.

### Expected behavior
A controller must add the finalizer, the same way it also removes it during cleanup. The KKP API must not set finalizers or owner references.

### PR DESCRIPTION
**Pull request by imharshita**
Signed-off-by: Harshita sharma <harshita.sharma6174@gmail.com>

**What does this PR do / Why do we need it**: Remove finalizers handling from api handler for externalclusters

**Does this PR close any issues?**:
Fixes #10376

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: pkg/controller/master-controller-manager/external-cluster/cluster_controller.go [hunk 1]
```diff
@@ -75,13 +77,34 @@ func Add(
...
+}
+
+func withEventFilter() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			externalCluster, ok := e.Object.(*kubermaticv1.ExternalCluster)
+			if !ok {
+				return false
+			}
+			return externalCluster.Spec.CloudSpec.KubeOne == nil
```
Comment (xrstf) [inline pkg/controller/master-controller-manager/external-cluster/cluster_controller.go]: Why do we ignore KubeOne clusters during their creation, but reconcile them when they are updated or deleted?

File: pkg/controller/master-controller-manager/external-cluster/cluster_controller.go [hunk 2]
```diff
@@ -75,13 +77,34 @@ func Add(
...
 }
 
 func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	resourceName := request.Name
-	log := r.log.With("request", request)
-	log.Debug("Processing")
+	log := r.log.With("externalcluster", request)
+	log.Debug("Processing...")
 
 	icl := &kubermaticv1.ExternalCluster{}
```
Comment (xrstf) [inline pkg/controller/master-controller-manager/external-cluster/cluster_controller.go]: Can we find a better variable name than "icl" for an external cluster?

**Review by xrstf — APPROVED** [review]
/approve

**Review by pkprzekwas — APPROVED** [review]

**Pull request merged**